source/ota.c

```diff
@@ -1772,62 +1580,62 @@ static OtaJobParseErr_t parseJobDocFromCustomCallback( const char * pJson,
 {
     OtaErr_t otaErr = OTA_ERR_NONE;
 
-    /* We have an unknown job parser error. Check to see if we can pass control to a callback for parsing */
-    OtaJobParseErr_t err = otaAgent.palCallbacks.customJobCallback( pJson, messageLength );
+    OtaJobParseErr_t err = OtaJobParseErrNone;
 
-    if( err == OtaJobParseErrNone )
+    /* We have an unknown job parser error. Check to see if we can pass control to a callback for parsing */
+    if( otaAgent.customJobCallback != NULL )
     {
-        /* Custom job was parsed by external callback successfully. Grab the job name from the file
-         *  context and save that in the ota agent */
-        if( pFileContext->pJobName != NULL )
-        {
-            otaAgent.pOtaSingletonActiveJobName = pFileContext->pJobName;
-            pFileContext->pJobName = NULL;
-            otaErr = otaControlInterface.updateJobStatus( &otaAgent,
-                                                          JobStatusSucceeded,
-                                                          JobReasonAccepted,
-                                                          0 );
+        err = otaAgent.customJobCallback( pJson, messageLength );
 
-            if( otaErr != OTA_ERR_NONE )
+        if( err == OtaJobParseErrNone )
+        {
+            /* Custom job was parsed by external callback successfully. Grab the job name from the file
+             *  context and save that in the ota agent */
+            if( strlen( pFileContext->pJobName ) > 0 )
             {
-                LogError( ( "Failed to update job status: "
-                            "updateJobStatus returned error: "
-                            "OtaErr_t=%d",
-                            otaErr ) );
-            }
+                memcpy( otaAgent.pActiveJobName, otaAgent.fileContext.pJobName, OTA_JOB_ID_MAX_SIZE );
```

Comment:
If you're going to count the string length, shouldn't you use that here instead of OTA_JOB_ID_MAX_SIZE?

Reply:
Updated all three occurrences 7d00faf